File: backend/functions/src/lib/firestorePaths.ts
export const paths = {
  user: (uid: string) => `users/${uid}`,
  group: (groupId: string) => `groups/${groupId}`,
  groupMember: (groupId: string, uid: string) => `groups/${groupId}/members/${uid}`,
  groupLogs: (groupId: string) => `groups/${groupId}/logs`,
  inviteCode: (code: string) => `inviteCodes/${code}`,
  leaderboardDaily: (groupId: string, dateKey: string) =>
    `groups/${groupId}/leaderboardDaily/${dateKey}`,
  leaderboardWeekly: (groupId: string, weekKey: string) =>
    `groups/${groupId}/leaderboardWeekly/${weekKey}`,
  badge: (groupId: string, badgeId: string) => `groups/${groupId}/badges/${badgeId}`,
};

File: ios/PeeTracker/PeeTracker/Firebase/FirestorePaths.swift
import FirebaseFirestore

enum FirestorePaths {
    static func user(_ uid: String) -> DocumentReference {
        Firestore.firestore().collection("users").document(uid)
    }

    static func group(groupId: String) -> DocumentReference {
        Firestore.firestore().collection("groups").document(groupId)
    }

    static func groupMember(groupId: String, uid: String) -> DocumentReference {
        group(groupId: groupId).collection("members").document(uid)
    }

    static func groupLogs(groupId: String) -> CollectionReference {
        group(groupId: groupId).collection("logs")
    }

    static func groupLog(groupId: String, logId: String) -> DocumentReference {
        groupLogs(groupId: groupId).document(logId)
    }

    static func leaderboardDaily(groupId: String, dateKey: String) -> DocumentReference {
        group(groupId: groupId).collection("leaderboardDaily").document(dateKey)
    }

    static func leaderboardWeekly(groupId: String, weekKey: String) -> DocumentReference {
        group(groupId: groupId).collection("leaderboardWeekly").document(weekKey)
    }

    static func groupBadges(groupId: String) -> CollectionReference {
        group(groupId: groupId).collection("badges")
    }
}
